libyul/AsmParser.cpp

```diff
@@ -175,7 +175,8 @@ Statement Parser::parseStatement()
 	case Token::Comma:
 	case Token::AssemblyAssign:
 	{
-		std::vector<Identifier> variableNames;
+		Assignment assignment;
+		assignment.location = locationOf(elementary);
```

Comment:
If I understand the change correctly, location of the assignment used to be set to:
```
abc, def, ghi := xyz
          ^^^
```
and now it will be
```
abc, def, ghi := xyz
^^^
```
is that the intention here? Why not
```
abc, def, ghi := xyz
^^^^^^^^^^^^^
```
or
```
abc, def, ghi := xyz
^^^^^^^^^^^^^^^^^^^^
```
?

Reply:
the source location is the full assignment statement now. I will add a syntax test.

Reply:
I marked it as approved since the test clearly confirms that the location is indeed the whole assignment. But I'm still trying to figure out why. `elementary` is just a single `identifier` so `locationOf(elementary)` should be just the location of that identifier, right?

Reply:
Because of this line: https://github.com/ethereum/solidity/pull/9225/files#diff-cfa1a0f58085cfb54feda51c7f9d53fdR214

So before this change, the location was actually
```
abc, def, ghi := xyz
          ^^^^^^^^^^
```

Reply:
Right, it overwrites the `end` later. I totally missed that.